framework/src/Volo.Abp.BlobStoring.Azure/Volo/Abp/BlobStoring/Azure/AzureBlobProvider.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+using Volo.Abp.DependencyInjection;
+
+namespace Volo.Abp.BlobStoring.Azure
+{
+    public class AzureBlobProvider : BlobProviderBase, ITransientDependency
+    {
+        protected IAzureBlobNameCalculator AzureBlobNameCalculator { get; }
+
+        public AzureBlobProvider(IAzureBlobNameCalculator azureBlobNameCalculator)
+        {
+            AzureBlobNameCalculator = azureBlobNameCalculator;
+        }
+
+        public override async Task SaveAsync(BlobProviderSaveArgs args)
```

Comment:
You never use args.ContainerName. Should we use it if `AzureBlobProviderConfiguration.ContainerName` was not specified?

Reply:
Although their names are the same (`ContainerName`), there is still a difference. The container name of Azure should conform to a little rule.

- This name may only contain lowercase letters, numbers, and hyphens, and must begin with a letter or a number.
- Each hyphen must be preceded and followed by a non-hyphen character.
- The name must also be between 3 and 63 characters long.

I think the `ContainerName `of `AzureBlobProviderConfiguration `must be specified. If it does not exist in azure, create the container. (`CreateContainerIfNotExists`)

Reply:
I have updated the code.

`Use ContainerName of the BlobProviderArgs if ContainerName of the AzureBlobProviderConfiguration is not specified.`

There are some rules for naming Azure container, which we need to explain in the documentation.

- This name may only contain lowercase letters, numbers, and hyphens, and must begin with a letter or a number.
- Each hyphen must be preceded and followed by a non-hyphen character.
- The name must also be between 3 and 63 characters long.

Reply:
Thanks. The documentation nicely explains it. We will also take care to follow this rules for container names to not have a problem in case of developer uses azure but not specify the container name.